src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedConditionalOperatorTests.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    /// <summary>
+    /// Test binding of the target-typed conditional (aka ternary) operator.
+    /// </summary>
+    public class TargetTypedConditionalOperatorTests : CSharpTestBase
+    {
+        [Fact]
+        public void TestImplicitConversions_Good()
+        {
+            // NOTE: Some of these are currently error cases, but they would become accepted (non-error) cases
+            // if we extend the spec to permit target typing even when there is a natural type.  Until then,
+            // they are error cases but included here for convenience.
+
+            // Implicit constant expression conversions
+            TestConditional("b ? 1 : 2", "System.Int16", "System.Int32",
```

Comment:
Could you add a few regression tests for cases where there is no target-type? Cases that may be interesting:

```C#
var x = b ? 1 : 2; // x : int
var x = b ? GetB() : GetC() // error, no common type
var x = b ? $""x"" : $""x"" // x : string
```

Reply:
This file adds target-typing tests. Existing scenarios with a var-declared variable are tested in `ConditionalOperatorTests.cs`. See line 28 there for the first case.
The second scenario is tested at line 95.
Line 119 demonstrates the third case.


---
In reply to: [412552731](https://github.com/dotnet/roslyn/pull/40604#discussion_r412552731) [](ancestors = 412552731)